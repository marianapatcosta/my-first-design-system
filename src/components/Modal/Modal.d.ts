import React, { CSSProperties, MouseEventHandler } from 'react';
export declare type ModalSize = 'small' | 'medium' | 'large';
export interface ModalProps {
    /**
     * element id
     */
    id?: string;
    /**
     * additional css class
     */
    className?: string;
    /**
     * additional style
     */
    style?: CSSProperties;
    /**
     * How large should the Modal be?
     */
    size?: ModalSize;
    /**
     * Modal's header
     */
    header?: string;
    /**
     * Modal's message content
     */
    message: string;
    /**
     * Label to display in Modal's button
     */
    buttonLabel: string;
    /**
     * Is Modal for confirm some action? It display 2 buttons for confirm and cancel action
     */
    isConfirmationModal?: boolean;
    /**
     * Label to display in confirmation modal, if isConfirmationModal = true
     */
    confirmationButtonLabel?: string;
    /**
     * Click handler to call on close ou on cancel (if isConfirmationModal = true)
     */
    onClose: MouseEventHandler<HTMLButtonElement>;
    /**
     * Click handler to call on confirmation, if isConfirmationModal = true
     */
    onConfirm?: MouseEventHandler<HTMLButtonElement>;
}
export declare const Modal: React.FC<ModalProps>;
