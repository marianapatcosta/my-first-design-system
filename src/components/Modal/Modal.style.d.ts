import { MouseEventHandler } from 'react';
import { ModalSize } from './Modal';
export declare const StyledModalOverlay: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const StyledModalContent: import("styled-components").StyledComponent<"div", any, {
    size?: ModalSize | undefined;
}, never>;
export declare const StyledModalHeader: import("styled-components").StyledComponent<"header", any, {}, never>;
export declare const StyledModalHeaderTitle: import("styled-components").StyledComponent<"h2", any, {}, never>;
export declare const StyledModalHeaderClose: import("styled-components").StyledComponent<import("react").ForwardRefExoticComponent<import("../Button/Button").ButtonProps & import("react").RefAttributes<HTMLButtonElement>>, any, {
    icon?: string | undefined;
    onClick?: MouseEventHandler<HTMLButtonElement> | undefined;
}, never>;
export declare const StyledModalMessage: import("styled-components").StyledComponent<"p", any, {}, never>;
export declare const StyledModalFooter: import("styled-components").StyledComponent<"footer", any, {}, never>;
export declare const StyledModalFooterButton: import("styled-components").StyledComponent<import("react").ForwardRefExoticComponent<import("../Button/Button").ButtonProps & import("react").RefAttributes<HTMLButtonElement>>, any, {
    label?: string | undefined;
    isConfirmationModal?: boolean | undefined;
    onClick: MouseEventHandler<HTMLButtonElement>;
}, never>;
