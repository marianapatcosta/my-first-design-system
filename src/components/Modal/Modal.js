var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useEffect, useRef, } from 'react';
import { Close } from '../../assets/icons';
import { StyledModalOverlay, StyledModalContent, StyledModalHeader, StyledModalHeaderTitle, StyledModalHeaderClose, StyledModalMessage, StyledModalFooter, StyledModalFooterButton, } from './Modal.style';
export var Modal = function (_a) {
    var header = _a.header, message = _a.message, buttonLabel = _a.buttonLabel, confirmationButtonLabel = _a.confirmationButtonLabel, isConfirmationModal = _a.isConfirmationModal, size = _a.size, onClose = _a.onClose, onConfirm = _a.onConfirm, props = __rest(_a, ["header", "message", "buttonLabel", "confirmationButtonLabel", "isConfirmationModal", "size", "onClose", "onConfirm"]);
    var modalRef = useRef(null);
    var modalButtonRef = useRef(null);
    useEffect(function () {
        // !!modalButtonRef.current && modalButtonRef.current.focus();
        var handleClickOutside = function (event /* MouseEvent */) {
            var element = event.target;
            if (modalRef.current && !(modalRef === null || modalRef === void 0 ? void 0 : modalRef.current.contains(element))) {
                event.preventDefault();
                event.stopPropagation();
                !!onClose && onClose(event);
            }
        };
        document.addEventListener('click', handleClickOutside);
        return function () { return document.removeEventListener('click', handleClickOutside); };
    }, [modalRef, onClose]);
    return (_jsx(StyledModalOverlay, { children: _jsxs(StyledModalContent, __assign({ ref: modalRef }, props, { size: size }, { children: [_jsxs(StyledModalHeader, { children: [_jsx(StyledModalHeaderTitle, { children: header }, void 0),
                        _jsx(StyledModalHeaderClose, { "aria-label": 'close modal', icon: Close, onClick: onClose }, void 0)] }, void 0),
                _jsx("main", { children: _jsx(StyledModalMessage, { children: message }, void 0) }, void 0),
                _jsxs(StyledModalFooter, { children: [isConfirmationModal && confirmationButtonLabel && onConfirm && (_jsx(StyledModalFooterButton, { size: 'small', label: confirmationButtonLabel, onClick: onConfirm }, void 0)),
                        _jsx(StyledModalFooterButton, { size: 'small', isConfirmationModal: isConfirmationModal, ref: modalButtonRef, label: buttonLabel, onClick: onClose }, void 0)] }, void 0)] }), void 0) }, void 0));
};
Modal.defaultProps = {
    id: '',
    className: '',
    style: {},
    header: '',
    message: '',
    buttonLabel: '',
    size: 'medium',
    confirmationButtonLabel: '',
    isConfirmationModal: false,
    onClose: function () { return null; },
    onConfirm: function () { return null; },
};
