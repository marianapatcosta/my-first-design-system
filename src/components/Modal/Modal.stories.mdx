import { Meta } from "@storybook/addon-docs/blocks";
import { Modal } from "./Modal";

<Meta title="Design System|Modal" component=Modal />

# Modal


## Usage
- Default Modal:
  <Story name='Default'>
    <Modal />
  </Story>


## Other important notes
not applicable.