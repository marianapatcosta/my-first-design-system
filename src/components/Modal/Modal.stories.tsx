import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { Modal, ModalProps } from './Modal';

export default {
  title: 'Modal',
  component: Modal,
  args: {},
} as Meta;

const Template: Story<ModalProps> = (args) =>
  <Modal {...args} />;

  export const Default = Template.bind({});
  Default.args = {

  };