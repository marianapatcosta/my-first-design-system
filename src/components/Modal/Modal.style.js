var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled, { css } from 'styled-components';
import { baseStyles, colors } from '../../styles';
import { hexToRGB } from '../../utils/utils';
import { Button } from '../Button/Button';
var sizeStyles = {
    small: css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    width: 20rem;\n  "], ["\n    width: 20rem;\n  "]))),
    medium: css(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    width: 35rem;\n  "], ["\n    width: 35rem;\n  "]))),
    large: css(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n    width: 50rem;\n  "], ["\n    width: 50rem;\n  "]))),
};
export var StyledModalOverlay = styled.div(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  ", "\n  height: 100vh;\n  width: 100vw;\n  position: fixed;\n  top: 0;\n  left: 0;\n  background-color: ", ";\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  z-index: 100;\n"], ["\n  ", "\n  height: 100vh;\n  width: 100vw;\n  position: fixed;\n  top: 0;\n  left: 0;\n  background-color: ", ";\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  z-index: 100;\n"])), baseStyles, hexToRGB(colors.shadow, 0.9));
export var StyledModalContent = styled.div(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n  position: fixed;\n  top: 50%;\n  left: 50%;\n  ", ";\n  background-color: ", ";\n  box-shadow: 0 0.125rem 0.5rem ", ";\n  border-radius: 0.5rem;\n  transform: translate(-50%, -50%);\n"], ["\n  position: fixed;\n  top: 50%;\n  left: 50%;\n  ", ";\n  background-color: ", ";\n  box-shadow: 0 0.125rem 0.5rem ", ";\n  border-radius: 0.5rem;\n  transform: translate(-50%, -50%);\n"])), function (_a) {
    var size = _a.size;
    return size && sizeStyles[size];
}, colors.primary, colors.shadow);
export var StyledModalHeader = styled.header(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  padding: 0.3rem;\n  background-color: ", ";\n  color: ", ";\n  z-index: 10;\n  border-top-left-radius: 0.5rem;\n  border-top-right-radius: 0.5rem;\n"], ["\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  padding: 0.3rem;\n  background-color: ", ";\n  color: ", ";\n  z-index: 10;\n  border-top-left-radius: 0.5rem;\n  border-top-right-radius: 0.5rem;\n"])), colors.secondary, colors.highlight);
export var StyledModalHeaderTitle = styled.h2(templateObject_7 || (templateObject_7 = __makeTemplateObject(["\n  margin: 0.5rem;\n  font-weight: 700;\n  font-size: 100%;\n"], ["\n  margin: 0.5rem;\n  font-weight: 700;\n  font-size: 100%;\n"])));
export var StyledModalHeaderClose = styled(Button)(templateObject_8 || (templateObject_8 = __makeTemplateObject(["\n  border: none;\n  img {\n    ", ";\n    transform: rotate(0deg);\n    transition: transform 0.3s ease-in-out;\n\n    &:hover {\n      transform: rotate(90deg);\n    }\n  }\n"], ["\n  border: none;\n  img {\n    ", ";\n    transform: rotate(0deg);\n    transition: transform 0.3s ease-in-out;\n\n    &:hover {\n      transform: rotate(90deg);\n    }\n  }\n"])), colors.icon);
export var StyledModalMessage = styled.p(templateObject_9 || (templateObject_9 = __makeTemplateObject(["\n  padding: 1rem;\n  text-align: center;\n"], ["\n  padding: 1rem;\n  text-align: center;\n"])));
export var StyledModalFooter = styled.footer(templateObject_10 || (templateObject_10 = __makeTemplateObject(["\n  padding: 1.25rem 0.3rem;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n"], ["\n  padding: 1.25rem 0.3rem;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n"])));
export var StyledModalFooterButton = styled(Button)(templateObject_11 || (templateObject_11 = __makeTemplateObject(["\n  background-color: ", ";\n  margin: 0 0.625rem;\n\n  ", "\n"], ["\n  background-color: ", ";\n  margin: 0 0.625rem;\n\n  ",
    "\n"])), colors.transparent, function (_a) {
    var isConfirmationModal = _a.isConfirmationModal;
    return isConfirmationModal &&
        "\n    border-color: " + colors.red + ";\n    color: " + colors.red + ";\n  ";
});
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6, templateObject_7, templateObject_8, templateObject_9, templateObject_10, templateObject_11;
