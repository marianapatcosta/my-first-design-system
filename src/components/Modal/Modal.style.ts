import styled from "styled-components";
import { baseStyles, disabledStyles, colors, typography } from '../../styles';

export const StyledModal = styled.div`
  ${baseStyles}
`;