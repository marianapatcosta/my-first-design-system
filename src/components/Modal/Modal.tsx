import React, { CSSProperties } from 'react';
import { StyledModal } from './Modal.style';

export interface ModalProps {
  /**
   * element id 
   */
  id?: string;
  /**
   * additional css class
   */
  className?: string;
  /**
   * additional style
   */
  style?: CSSProperties;
};

export const Modal: React.FC<ModalProps>= ({ ...props}) => {
  return (
    <StyledModal  {...props}>
      
    </StyledModal>
  );
};


Modal.defaultProps = {
  id: '',
  className: '',
  style: {}
};