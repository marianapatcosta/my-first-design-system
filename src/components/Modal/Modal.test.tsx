import React from 'react';
import { render } from '@testing-library/react';
import { Modal } from './Modal';

describe('Modal', () => {
  const defaultProps = {};
   test('renders without errors and matches snapshot', () => {
     const { container: { firstChild }  } = render(<Modal  {...defaultProps}/>); 

     expect(firstChild).toMatchSnapshot();
   });
});