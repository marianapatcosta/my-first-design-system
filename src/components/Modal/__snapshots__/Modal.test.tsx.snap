// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders without errors and matches snapshot 1`] = `
<div
  class="sc-bdfBwQ cUiAAY"
  id=""
/>
`;
